rain_api_core/egress_util.py

```diff
@@ -150,32 +144,35 @@ def process_request(varargs, b_map):
     return path, bucket, object_name, headers
 
 
-def check_private_bucket(bucket, b_map):
-
+def check_private_bucket(bucket, b_map, optional_uri=""):
     log.debug('check_private_buckets(): bucket: {}'.format(bucket))
 
     # Check public bucket file:
     if 'PRIVATE_BUCKETS' in b_map:
         for priv_bucket in b_map['PRIVATE_BUCKETS']:
-            if bucket == prepend_bucketname(priv_bucket):
-                # This bucket is PRIVATE, return group!
-                return b_map['PRIVATE_BUCKETS'][priv_bucket]
+            if bucket_prefix_ismatch(bucket, priv_bucket, optional_uri):
+                    # This bucket is PRIVATE, return group!
+                    return b_map['PRIVATE_BUCKETS'][priv_bucket]
 
     return False
 
 
-def check_public_bucket(bucket, b_map):
-
+def check_public_bucket(bucket, b_map, optional_uri=""):
     # Check for PUBLIC_BUCKETS in bucket map file
     if 'PUBLIC_BUCKETS' in b_map:
         log.debug('we have a PUBLIC_BUCKETS in the ordinary bucketmap file')
         for pub_bucket in b_map['PUBLIC_BUCKETS']:
-            #log.debug('is {} the same as {}?'.format(bucket, prepend_bucketname(pub_bucket)))
-            if bucket == prepend_bucketname(pub_bucket):
+            # log.debug('is {} the same as {}?'.format(bucket, prepend_bucketname(pub_bucket)))
+            if bucket_prefix_ismatch(bucket, pub_bucket, optional_uri):
                 # This bucket is public!
                 log.debug('found a public, we\'ll take it')
                 return True
 
     # Did not find this in public bucket list
     log.debug('we did not find a public bucket for {}'.format(bucket))
     return False
+
+
+def bucket_prefix_ismatch(bucket_check, bucket_map, optional_uri):
+    bucket = prepend_bucketname(bucket_check)
```

Comment:
Is this the right one to prefix?

Reply:
Are you suggesting that the public bucket doesn't need to be pre-fixed?